cli/mender-artifact/dump.go

```diff
@@ -0,0 +1,242 @@
+		if err != nil {
+			return cli.NewExitError(fmt.Sprintf(
+				"Unable to create directory: %s", err.Error()), errSystemError)
+		}
+
+		// Hardcode to 0 index for now.
+		handler := handlers[0]
+
+		for _, augmented := range []bool{false, true} {
+			var metaData map[string]interface{}
+			var fullPath string
+			var metaDataArg string
+			if augmented {
+				metaData = handler.GetUpdateAugmentMetaData()
+				fullPath = path.Join(c.String("meta-data"), "0000.meta-data-augment")
+				metaDataArg = "--augment-meta-data"
+			} else {
+				metaData = handler.GetUpdateOriginalMetaData()
+				fullPath = path.Join(c.String("meta-data"), "0000.meta-data")
+				metaDataArg = "--meta-data"
+			}
+
+			if len(metaData) == 0 {
+				continue
+			}
+
+			metaDataFd, err := os.OpenFile(fullPath,
+				os.O_WRONLY|os.O_CREATE|os.O_EXCL, 0644)
+			if err != nil {
+				return cli.NewExitError(fmt.Sprintf(
+					"Unable to create meta-data file: %s", err.Error()), errSystemError)
+			}
+			defer metaDataFd.Close()
+
+			w := json.NewEncoder(metaDataFd)
+			err = w.Encode(metaData)
+			if err != nil {
+				return errors.New("Unencodeable map in dumpPayloads. Should not happen.")
+			}
+
+			dump.args = append(dump.args, metaDataArg, fullPath)
+		}
+	}
+
+	if len(c.String("files")) > 0 {
+		store := &dumpFileStore{
+			fileDir: c.String("files"),
+			args:    &dump.args,
+		}
+		for _, h := range handlers {
+			h.SetUpdateStorerProducer(store)
+		}
+	}
+
+	err := ar.ReadArtifactData()
+	if err != nil {
+		return cli.NewExitError(fmt.Sprintf("Error dumping Artifact: %s",
+			err.Error()), errArtifactInvalid)
+	}
+
+	return nil
+}
+
+func printCmdline(ar *areader.Reader, args []string) {
+	// Even if it is a rootfs payload, we use the module-image writer, since
+	// this can recreate either type.
+	fmt.Printf("write module-image")
+
+	artProvs := ar.GetArtifactProvides()
+	fmt.Printf(" --artifact-name %s", artProvs.ArtifactName)
+	if len(artProvs.ArtifactGroup) > 0 {
+		fmt.Printf(" --provides-group %s", artProvs.ArtifactGroup)
+	}
+
+	artDeps := ar.GetArtifactDepends()
+	if len(artDeps.ArtifactName) > 0 {
+		fmt.Printf(" --artifact-name-depends %s", strings.Join(artDeps.ArtifactName, " --artifact-name-depends "))
+	}
+	fmt.Printf(" --device-type %s", strings.Join(artDeps.CompatibleDevices, " --device-type "))
+	if len(artDeps.ArtifactGroup) > 0 {
+		fmt.Printf(" --depends-groups %s", strings.Join(artDeps.ArtifactGroup, " --depends-groups "))
+	}
+
+	handlers := ar.GetHandlers()
+	if len(handlers) > 0 {
+		fmt.Printf(" --type %s", ar.GetHandlers()[0].GetUpdateType())
+	}
+
+	if len(args) > 0 {
+		fmt.Printf(" %s\n", strings.Join(args, " "))
+	}
+}
+
+func (d *dumpFileStore) NewUpdateStorer(updateType string, payloadNum int) (handlers.UpdateStorer, error) {
+	return d, nil
+}
+
+func (d *dumpFileStore) Initialize(artifactHeaders,
+	artifactAugmentedHeaders artifact.HeaderInfoer,
+	payloadHeaders handlers.ArtifactUpdateHeaders) error {
+
```

Comment:
Space necessary here?

Reply:
I think it's clearer. `go fmt` never indents more than one level, so the parameters and the first statement look very similar, and it's hard to see where the code starts.